src/index.ts

```diff
@@ -48,68 +51,81 @@ function createRdfEmitter(program: Program) {
         if (doc) {
           writer.addQuad(
             nameNode,
-            nn("rdfs:comment"),
+            nn("skos:note"),
             DataFactory.literal(doc)
           );
         }
-        for (const prop of m.properties.values()) {
+
+        
+        // Data properties
+        for (const prop of m.properties.values()) 
+        {
           const propNameNode = nn(nameForProperty(prop));
-          writer.addQuad(propNameNode, nn("rdf:type"), nn("rdf:Property"));
+          writer.addQuad(propNameNode, nn("http://www.w3.org/1999/02/22-rdf-syntax-ns#type"), nn("owl:DataTypeProperty"));
           writer.addQuad(propNameNode, nn("rdfs:domain"), nameNode);
 
-          if (prop.type.kind === "Model") {
-            if (isArrayModelType(program, prop.type)) {
-              // after years of research I have been unable to determine how
-              // to create a subtype of seq with a particular element type
-              writer.addQuad(propNameNode, nn("rdfs:range"), nn("rdf:Seq"));
-            } else {
-              writer.addQuad(
+          if (prop.type.kind === "Model") 
+          {
+              writer.addQuad
+              (
                 propNameNode,
                 nn("rdfs:range"),
                 nn(nameForModel(prop.type))
               );
-            }
-          } else if (prop.type.kind === "Union") {
-            for (const variant of prop.type.variants.values()) {
-              if (variant.type.kind === "Model") {
-                writer.addQuad(
-                  propNameNode,
-                  nn("rdfs:rangeIncludes"),
-                  nn(nameForModel(variant.type))
-                );
-              } else if (
-                variant.type.kind === "String" ||
-                variant.type.kind === "Number"
-              ) {
-                writer.addQuad(
-                  propNameNode,
-                  nn("rdfs:rangeIncludes"),
-                  DataFactory.literal(variant.type.value)
-                );
+          }
+
+          else if (prop.type.kind === "Union") 
+          {
+
+            const arr= [];
+
+            for (const variant of prop.type.variants.values()) 
+            {
+
+              if (variant.type.kind === "Model") 
+              {
+                arr.push(nameForModel(variant.type));
+              }
+              if (variant.type.kind === "String" || variant.type.kind === "Number") 
+              {
+                arr.push(variant.type.value);
               }
-              // todo: support booleans and other exotic union types
             }
+    
+            writer.addQuad(propNameNode, nn("rdfs:range"), writer.blank([{
+              predicate: nn("http://www.w3.org/1999/02/22-rdf-syntax-ns#type"),
+              object:    nn("rdfs:Datatype"),
+            },{
+              predicate: nn("owl:oneOf"),
+              //TODO: owl:oneOf "red,blue" -> owl:oneOf ("red" "blue" )
+              object:  DataFactory.literal(arr.toString()),

```

Comment:
The reason you're getting the undesirable output is that you're making a literal of `arr.toString()` which results in the string `"red,blue"`. I think the key to making the output that you want is to use `writer.list`. If you construct `arr` (which should use a better name!) such that the array contains namedNodes or literals rather than just string values, and then on this line set `object` to `writer.list(arr)`, I think it will work.